src/apps/explorer/layout/Header.tsx

```diff
@@ -25,45 +30,17 @@ export const Header: React.FC = () => {
 
   const prefixNetwork = PREFIX_BY_NETWORK_ID.get(networkId)
 
-  const handleNavigate = (e: React.MouseEvent<HTMLAnchorElement>): void => {
-    e.preventDefault()
-    setBarActive(false)
-    history.push(`/${prefixNetwork || ''}`)
-  }
-
   return (
-    <GenericHeader logoAlt="CoW Protocol Explorer" linkTo={`/${prefixNetwork || ''}`}>
+    <GenericHeader
+      logoAlt="CoW Protocol Explorer"
+      linkTo={`/${prefixNetwork || ''}`}
+      onClickOptional={handleMobileMenuOnClick}
```

Comment:
Not sending the click function when the menu is closed would avoid opening the mobile menu when clicking on the logo.

```typescript
onClickOptional={isMobileMenuOpen ? handleMobileMenuOnClick : undefined}
```